test/e2e/plugin_lifecycle/plugin_lifecycle_test.go

```diff
@@ -5,39 +5,209 @@
 package plugin
 
 import (
+	"fmt"
+	"os"
+
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 
 	"github.com/vmware-tanzu/tanzu-cli/test/e2e/framework"
 )
 
+// This test suite covers plugin life cycle use cases for central repository
+// it uses local central repo to discovery plugins, for which we need to make sure that
+// docker is running and also local central repo is running, start with 'make start-test-central-repo'
+// we need to update PATH with tanzu binary
+// run the tests with make target 'make start-test-central-repo',
+// this make target by default updates the local central repository URL to environment variable TANZU_CLI_E2E_TEST_LOCAL_CENTRAL_REPO_URL
+// the use case being covered in this suite are:
+// 1. plugin search, install, delete, describe, list (with negative use cases)
+// 2. plugin source add/update/list/delete (with negative use cases)
 var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Plugin-lifecycle]", func() {
 	var (
-		tf               *framework.Framework
-		plugins          []*framework.PluginMeta
-		discoveryMap     map[string]string
-		discoverySources []string
+		tf                         *framework.Framework
+		e2eTestLocalCentralRepoURL string
+		pluginSourceName           string
+		pluginsSearchList          *[]framework.PluginInfo
 	)
 	BeforeSuite(func() {
 		tf = framework.NewFramework()
-		pm := framework.NewPluginMeta()
-		plugins = make([]*framework.PluginMeta, 1)
-		pm.SetName("dummy").SetTarget("k8s").SetVersion("1.0").SetDescription("its dummy plugin").SetSHA("345").SetGroup("admin").SetArch("amd64").SetOS("darwin").SetDiscoveryType("oci").SetHidden(false).SetOptional(false)
-		plugins[0] = pm
-		GenerateAndPublishScriptBasedPluginsToLocalOCIRepo(tf, plugins[:])
+
+		// check E2E test central repo URL
+		e2eTestLocalCentralRepoURL = os.Getenv(framework.TanzuCliE2ETestLocalCentralRepositoryURL)
+		Expect(e2eTestLocalCentralRepoURL).NotTo(BeEmpty(), fmt.Sprintf("environment variable %s should set with local central repository URL", framework.TanzuCliE2ETestLocalCentralRepositoryURL))
+		// set E2E test central repo URL to TANZU_CLI_PRE_RELEASE_REPO_IMAGE
+		os.Setenv(framework.CentralRepositoryPreReleaseRepoImage, e2eTestLocalCentralRepoURL)
+
+		// set central repo feature flag
+		err := tf.Config.ConfigSetFeatureFlag(framework.CentralRepositoryFeatureFlag, framework.True)
```

Comment:
nit: Needs to be updated once #105 is merged.

Reply:
#105 was merged, so the above line can be removed